tests/integration/test_charm.py

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python3
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import asyncio
+import logging
+
+import pytest
+from helpers import get_prometheus_rules
+
+logger = logging.getLogger(__name__)
+
+APP_NAME = "cos-config"
+PROM_NAME = "prometheus"
+SPRING_NAME = "spring-music"
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test, charm_under_test):
+    await ops_test.model.deploy(charm_under_test, application_name=APP_NAME)
+    # The charm should be in blocked state if not related to anything.
+    await ops_test.model.wait_for_idle(status="blocked")
+
+
+async def test_relate(ops_test):
+    await asyncio.gather(
+        ops_test.model.deploy("ch:prometheus-k8s", application_name=PROM_NAME, channel="beta"),
+        ops_test.model.deploy("ch:spring-music", application_name=SPRING_NAME, channel="edge"),
+    )
```

Comment:
Consider moving the `deploy` to the first function.

Reply:
Why?

Reply:
- Increase clarity: all you need for this test can be deployed at the start, so it might be easier for the reader to understand all the actors at play.
- Follow up test funcs are (conveniently) named `test_relate` (and not `test_deploy_and_relate`).

This is just a suggestion so feel free to resolve.